nginx/ng/certificates.sls

```diff
@@ -1,6 +1,23 @@
 include:
   - nginx.ng.service
 
+{% if salt.pillar.get('nginx:ng:dh_contents') %}
+create_nginx_dhparam_key:
+  file.managed:
+    - name: /etc/nginx/ssl/dhparam.pem
+    - contents_pillar: nginx:ng:dh_contents
+    - makedirs: True
+{% elif salt.pillar.get('nginx:ng:dh_keygen', False) %}
+generate_nginx_dhparam_key:
+  file.directory:
+    - name: /etc/nginx/ssl
+    - makedirs: True
+  cmd.run:
+    - name: openssl dhparam -out dhparam.pem {{ salt.pillar.get('nginx:ng:dh_keysize', 2048) }}
```

Comment:
possibly allow them to specify pem name, if their config relies on a specific file or path